src/lib/ngsi/ContextElement.cpp

```diff
@@ -154,7 +157,7 @@ void ContextElement::fill(const struct ContextElement& ce)
   attributeDomainName.fill(ce.attributeDomainName);
   contextAttributeVector.fill((ContextAttributeVector*) &ce.contextAttributeVector);
   domainMetadataVector.fill((MetadataVector*) &ce.domainMetadataVector);
-  providingApplication = ce.providingApplication;  
+  providingApplicationList = ce.providingApplicationList;  // FIXME P10: not sure if this is the right way of doing, taking into account how the same is done for contextAttributeVector in its own fill() method...
```

Comment:
The class std::vector could implement the operator '=' as to do what you are proposing.
However, how can it know exactly what has to be done for each type of vector?
Imagine a ContextElement-vector, that has a ContextAttribute-vector inside, which in its turn has a Metadata-vector inside ...

Seems impossible to implement,
Perhaps it is possible in an interpreter, but a compiler ... I doubt it.

I think you will have to implement the 'cloning' yourself.